lib/sync.js

```diff
@@ -0,0 +1,51 @@
+import VP from "./vp";
+import {
+  getFlowsByStatus,
+  updateParliamentFlowStatus,
+} from "./parliament-flow";
+import { PARLIAMENT_FLOW_STATUSES, VP_PARLIAMENT_FLOW_STATUSES } from "../config";
+
+/**
+ * For all flows in Kaleidos that have a given status, it checks for
+ * status updates on the VP API and updates their status in
+ * Kaleidos accordingly.
+ */
+async function syncFlowsByStatus(statuses) {
+  const pendingFlows = await getFlowsByStatus(statuses);
+  await Promise.all(
+    pendingFlows.map(async (flow) => {
+      let statuses;
+      try {
+        statuses = await VP.getStatusForFlow(flow.parliamentId);
+      } catch (error) {
+        console.error(error);
+      }
+      if (!statuses?.statussen) {
+        await updateParliamentFlowStatus(
+          flow.uri,
+          PARLIAMENT_FLOW_STATUSES.VP_ERROR,
+          true
+        );
+        console.warn(
+          `Invalid response from status API for flow with uri ${flow.uri}`
+        );
```

Comment:
**question**: Will we fall into this if-case if the response is `{ statussen: null }`?

It seems that we don't get real responses for flows that are immediately made, they only get a status set once a user at the VP side picks up the flow for processing. In the meantime we get the `{ statussen: null }` response.

Reply:
I was not aware this is the case. We decided to not set any error status here, and check with VP what we should expect when an error happens.